### ISSUE TITLE
After upgrading to Flutter 3.10.1 and enabling FLTEnableWideGamut rear camera photos do not display at all 

### ISSUE THREAD

**Issue by jabedamin** [flutter/flutter#136629]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Create a function to select an image from your device gallery using image_picker (official flutter package)
3. Select an image taken with the rear camera of your device (using an iOS actual device - in this example iPhone 13 pro)
4. Either save the image to a server and retrieve the url or use File object to store the image.
5. Display the image in the appropriate widget (Image.network(url) or Image.file(path). 

### Expected results

Expected to display the image that was loaded from the device.

### Actual results

Any photo taken using the rear camera will not display. If FLTEnableWideGamut is enabled you'll see a bright yellowish/beige colour take up the space. 

If you screenshot to share this the area then just shows plain white on different screens. I believe this is something to do with the wide gamut support for iOS. 

You no longer need to manually enable the FLTEnableWideGamut but this problem started only since the initial support for this was enable in 3.10.1. In later versions it became default setting. A lot the code I wrote was before this feature was released and the code worked fine. 

Now there are no errors in the console but rear camera images just won't display (even if you upload them to a server then use the url or directly use the file path on the device). 

However, any image taken using the front facing camera, or saved from other apps or screenshots seem to render absolutely fine. It's very bizarre behaviour took me a long time to realise this difference 

See screenshot of issue below - FYI i have tested thoroughly and the data is loading fine, the image is there. It's just not rendering correctly if it originally was taken on the rear camera.

 

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';
import 'package:json_annotation/json_annotation.dart';
const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class ProfilePhotoWidget extends StatefulWidget {
  const ProfilePhotoWidget({
    super.key,
    required this.profileId,
  });

  final String profileId;
  final String currentPhotoUrl;

  @override
  State<ProfilePhotoWidget> createState() =>
      _ProfilePhotoWidgetState();
}

class _ProfilePhotoWidgetState extends State<ProfilePhotoWidget> {
 
  
  Future<void> updateProfilePhoto() async {
    final ImagePicker _picker = ImagePicker();
    final XFile? image = await _picker.pickImage(source: ImageSource.gallery);

    if (image == null) {
      return;
    }

    // upload image to Firebase Storage
    Reference storageReference = FirebaseStorage.instance
        .ref()
        .child('users/$profileId/profilePic/$profileId.jpg');

    final UploadTask uploadTask = storageReference.putFile(
        File(image.path), SettableMetadata(contentType: 'image/jpeg'));

    // Get the download URL once the image is uploaded
    await uploadTask.whenComplete(() async {
      final String profilePhotoUrl = await storageReference.getDownloadURL();

    // Update the Firestore document with the new profile photo URL
      FirebaseFirestore.instance.collection('users').doc(profileId).update({
        'profilePhotoUrl': profilePhotoUrl,
      }).then((_) {
        print('profile photo updated successfully.');
      }).catchError((error) {
        print('failed to update profile photo: $error');
      });
    });
  }


  @override
  Widget build(BuildContext context) {
    final screenSize = MediaQuery.of(context).size;
    return Align(
      alignment: Alignment.center,
      child: Stack(
        children: [
          Container(
            clipBehavior: Clip.hardEdge,
            width: screenSize.width / 2.5,
            height: screenSize.width / 2.5,
            decoration: BoxDecoration(
              borderRadius: BorderRadius.circular((screenSize.width / 2.5) / 2),
            ),
            child: Image.network(widget.currentPhotoUrl,
                fit: BoxFit.cover),
          ),
          
         
            Positioned(
              top: 0,
              right: 0,
              child: GestureDetector(
                onTap: () async {
                  print('did tap edit photo');
               
                  await updateProfilePhoto();
               
                },
                child: Container(
                  height: (screenSize.width / 2.5) * 0.23,
                  width: (screenSize.width / 2.5) * 0.23,
                  decoration: BoxDecoration(
                    color: Colors.white,
                    borderRadius:
                        BorderRadius.circular((screenSize.width * 0.23) / 2),
                  ),
                  child: Icon(
                    Icons.edit,
                    color: Colors.blue,
                  ),
                ),
              ),
            )
        ],
      ),
    );
  }
}





class ProfileView extends StatefulWidget {
  const ProfileView({Key? key, required this.businessId})
      : super(key: key);

  final String profileId;

  @override
  State<ProfileView> createState() =>
      _ProfileViewState();
}

class _ProfileViewState extends State<ProfileView> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return StreamBuilder<ProfileModel>(
        stream: FirebaseFirestore.instance
                                .collection('users')
                                .doc(profileId)
                                .snapshots();,
        builder: (context, snapshot) {
          if (snapshot.connectionState == ConnectionState.active) {
            final profile = ProfileModel.fromJson(snapshot.data! as Map<String,dynamic>);

            return Container(
              child: ListView(
                children: [
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      SectionHeader(label: 'PROFILE PHOTO'),
                      ProfilePhotoWidget(profileId: widget.profileId, currentPhotoUrl: profile.profilePhotoUrl),
                      
                    ],
                  ),
                ],
              ),
            );
          }
          return SizedBox();
        },);
  }
}


@JsonSerializable(explicitToJson: true)
class ProfileModel {
  String uid;
  String profilePhotoUrl;

  ProfileModel({
    this.uid,
    this.profilePhotoUrl
  });

  factory ProfileModel.fromJson(Map<String, dynamic> json) =>
      _$ProfileModelFromJson(json);

  Map<String, dynamic> toJson() => _$ProfileModelToJson(this);
}



```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]
![IMG_7339](https://github.com/flutter/flutter/assets/74027256/bd3ec4a8-6c30-4bfb-8e57-ff334292ae3a)

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.3.1 22xxxxxxxa darwin-arm64, locale en-GB)
    • Flutter version 3.10.5 on channel stable at /Users/jabed/Developer/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (4 months ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/jabed/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • Jabed XIII (mobile) • 0xxxxx-xxxxxxxx • ios            • iOS 16.6.1 20G81
    • macOS (desktop)     • macos                     • darwin-arm64   • macOS 13.3.1 22xxxxxx darwin-arm64
    • Chrome (web)        • chrome                    • web-javascript • Google Chrome 117.0.xxxx.xxx

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @jabedamin 
Your sample code is using 3rd plugin (firebase) so I check this issue with image_picker [example](https://github.com/flutter/packages/blob/main/packages/image_picker/image_picker/example/lib/main.dart). The issue doesn't occur on my device `iPhone 7, iOS 15.7.9` (rear camera) and on latest Flutter master channel.

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.13.7, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.13.7 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2f708eb839 (3 days ago), 2023-10-09 09:58:08 -0500
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 117.0.5938.149

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.16.0-11.0.pre.91, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.16.0-11.0.pre.91 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6074ff5e90 (11 minutes ago), 2023-10-16 22:53:32 -0400
    • Engine revision 2159fdb846
    • Dart version 3.3.0 (build 3.3.0-29.0.dev)
    • DevTools version 2.28.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.9 19H365
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 118.0.5993.70

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

Please confirm if you still see the issue with image_picker example on the latest Flutter channels. If it does, please provide device information as well (OS, model). Thanks!

**Comment by jabedamin** [comment]
Hi @huycozy,

The code sample is a simplified version of my actual code that I created in DartPad. In the actual code the Firebase packages are all imported correctly. Is that what you mean? Let me know if I've misinterpreted your question.

My device is an iPhone 13 Pro, iOS 16.6.1. One observation is that this code existed in the app before and worked fine until upgrading to Flutter 3.10.1 and onwards the issue has persisted.

I also ran the app on an iOS simulator via Xcode, can't test the camera on the simulator but the same images taken on the other device also don't load correctly there. Just shows plain white. 

Interestingly though, the same images do render correctly on my Android device (Samsung Galaxy A10). 

So in summary it seems to be that any photo taken with rear camera on the iPhone 13 pro uploads the image correctly to the servers but cannot display them on iOS devices but displaying them on Android devices works fine.

**Comment by huycozy** [comment]
>In the actual code the Firebase packages are all imported correctly. Is that what you mean?

I mean the sample code should be minimal (without using 3rd package) so it's easier for us to verify the correct issue. Firebase is 3rd package. Thus, we need to determine if it is necessary to reproduce this issue.

>Now there are no errors in the console but rear camera images just won't display (even if you upload them to a server then use the url or directly use the file path on the device).

Based on the given information, it seems the image taken by rear camera was unsuccessfully and you should isolate the issue with `image_picker` plugin only. You can try checking the issue with image_picker [example](https://github.com/flutter/packages/blob/main/packages/image_picker/image_picker/example/lib/main.dart) I provided above to see if the issue also occurs on your device. The image_picker example loads image to Image directly without uploading to Firebase. I also checked this on iPhone 14 Pro, iOS 17.0 and enabled Gamut but the image still can display well.

**Comment by jabedamin** [comment]
Okay, I tested with your code exactly. And the issue is still there consistent with the original description.

I took a photo of the screen from a different phone so you can see what it actually looks like. There's is a beige/yellowish colour that shows in place of the image. This happens with every single Rear Camera photo. If I take a screenshot instead of the yellow/beige it just shows plain white, also if i turn brightness to full on the device the colour turns just white.
![20231024_153050](https://github.com/flutter/flutter/assets/74027256/ff5d2278-bb53-41de-91c3-642b410f298c)

I don't think it is an image_picker issue because I used the same version of image_picker before upgrading to Flutter 3.10.1 and the same code in my app worked without this issue.

What do you think looking at the picture ?

**Comment by huycozy** [comment]
Perhaps this issue is an issue with image rendering rather than image_picker plugin. But to verify this for sure, could you try reproduce the issue in below scenario:

1. Capture an image by device's rear camera app
2. Copy/move image to a new Flutter project, in `assets` directory
3. Load image with `Image.asset`
4. Enable Gamut and run app to see if image is displayed or not

**Comment by jabedamin** [comment]
Steps taken:
1. Take a photo with rear camera on iPhone 13 pro
2. Transfer to mac by saving the photo to 'Files' app (in iCloud)- image stored as .HEIC
3. Add to assets of a new flutter project
4. Load with image.asset - this does work fine

So if i manually moved the photo from my iphone to icloud then transferred to assets in the flutter Project as .HEIC file, there is no issue.

Of course, in reality images won't be transferred to the flutter app in this way. Developers have no choice but to use Flutter's recommended image_picker package. So I did some further testing enabling/disabling the FLTEnableWideGamut flag in info.plist. 

**Important Observation - if I remove the FLTEnableWideGamut the images DO render correctly, but if this flag is enabled, the images DO NOT render** - this applies to my app and both of your example apps if I follow the recommended flutter approach with image_picker.

From my assessment it is definitely related to the FLTEnableWideGamut flag, but may also be related to Image_Picker. Also if you see the final paragraph in my previous comment, the only way to make it work with FLTEnableWideGamut flag enabled and using image_picker concurrently, is to specify maxWidth and maxLength (which should be optional values?)

So what do you think we should focus on?

**Comment by huycozy** [comment]
Thanks for checking that. I also haven't thought of any other scenario to check if `image_picker` is really the cause or just a factor to produce the issue. Labeling the issue for other's opinions.

**Comment by zanderso** [comment]
@huycozy Does this reproduce on 3.13 stable or 3.16 beta?

**Comment by huycozy** [comment]
@zanderso Unfortunately, I can't reproduce this issue on my device `iPhone 7, iOS 15.7.9` as mentioned above. I also asked my colleague to check this issue on his device `Iphone XS, iOS 17.1` but also can't reproduce this. 

@jabedamin Could you check this on your device?

**Comment by jabedamin** [comment]
@huycozy @zanderso - I can't test this immediately but in 2 weeks we will perform an upgrade. I'll retest this and feed back here.

**Comment by jabedamin** [comment]
@huycozy @zanderso positive update guys, not sure exactly what the fix was but after updating to flutter 3.13 the issue seems to have disappeared. 

Rear camera photos previously showing blank now appear correctly. Also no issue uploading new photos.

**Comment by jonahwilliams** [comment]
Closing as fixed
